### ISSUE TITLE
Problem in workflow with RabbitMQ configured with Parallel message processing limit equal to 1

### ISSUE THREAD

**Issue by cbalbinos** [n8n-io/n8n#8167]
Hi,

A customer reported a situation where RabbitMQ queue consumption is not respecting one execution at a time, even though the Parallel Message Processing Limit is set to 1

In the tests I carried out, I observed that this behavior occurs when the wait time is greater than 60 seconds, or even when the wait is defined in minutes

In the print below where it shows 5 executions at 18:06:07 the wait was set to 2 minutes, and the 5 executions below have different execution times with a wait of 30 seconds

From what I understand by reading reports from other issues, when wait is longer than 65 seconds it changes the execution status to waiting, considering that the workflow was executed successfully, even though it has not been executed yet

The tests were carried out with versions 1.15.2 and 1.19.0, installed in docker, queue mode and postgres

![Execuções](https://github.com/n8n-io/n8n/assets/91399681/f3ebb8b8-f8a6-4941-a389-7bf7832c00fd)

![config_RabbitMQ Trigger](https://github.com/n8n-io/n8n/assets/91399681/62ea4baa-62a8-4d82-ae7f-f3958028d26a)

**Comment by Joffcom** [comment]
Hey @cbalbinos,

It sounds like you are probably correct there, Whenever a workflow goes into the database in a waiting state (over 65 seconds on a wait) the triggering workflow sort of sits there in a pending state as we don't really know when the workflow would be finished it is considered partially complete so it doesn't block anything.

I am not sure if this is something that would be quick or easy to change to make as it part of the core execution logic rather than just an issue with a node not working correctly.

I have given this a quick test though and it looks like the workflow is still processing one item at a time and handles the messages in order. Is this actually causing an issue with the workflows or the processing of data?

**Comment by cbalbinos** [comment]
Hi @Joffcom ,

This is a problem because the same behavior is expected for workflows that contain a wait of 30 seconds or 2 minutes.

In my example, it consumed all items from the RabbitMQ queue at the same time, I expect it to take a new item from the queue only when the previous item is processed successfully.

**Comment by Joffcom** [comment]
Hey @cbalbinos,

Perfect so in your case you want to fully finish processing one item before moving to the next. I will get a dev ticket created for this but I don't think it will be fixed quickly, As a temporary solution I would use multiple wait nodes if you need to use them and keep them to under 65 seconds on each one.

**Comment by cbalbinos** [comment]
Ok @Joffcom,


To get around the problem in my use case, I started adding the RabbitMQ items to a Sheets spreadsheet, now I take the items from the spreadsheet and use wait with a time greater than 65 seconds in a loop

**Comment by Joffcom** [comment]
Hey @cbalbinos,

That is one way to do it although it kind of makes the queue useless and you might as well replace both RabbitMQ and the Sheet with a postgres table to achieve the same thing. Unless Rabbit is used for other things as well

**Comment by cbalbinos** [comment]
Yes @Joffcom , in this use case the client sends a large volume of information from Mautic to a Webhook, which is why the information is sent to RabbitMQ, and then to Sheets (in the scenario without consuming directly from the RabbitMQ queue) limiting RabbitMQ parallelism to not exceed the Sheets API limit
